diff --git a/index.d.ts b/index.d.ts
index 356f5503b1fd765e5c233b209ea76a84e134219f..bc35f070ef384c7a8424492a1f9ecd6c96804721 100755
--- a/index.d.ts
+++ b/index.d.ts
@@ -40,6 +40,15 @@ declare namespace Emscripten {
     }
 }
 
+/* fake dom types, for when dom globals are not used */
+interface MessageEventInit<T = any> {}
+interface MessageEvent<T = any> {}
+declare var MessageEvent: {
+  prototype: MessageEvent;
+  new<T>(type: string, eventInitDict?: MessageEventInit<T>): MessageEvent<T>;
+};
+/* end fake dom types */
+
 interface EmscriptenModule {
     print(str: string): void;
     printErr(str: string): void;
